hyperspy/axes.py

```diff
@@ -535,6 +489,68 @@ def __deepcopy__(self, memo):
         cp = self.copy()
         return cp
 
+    def _parse_value_from_string(self, value):
+        test_value = value if type(value) == str else value[0] # else array-like, and test first entry
```

Comment:
This doesn't look right to me: this function is expecting a string, and in case this is not a string, why would you take the first item and not any other?